### PR TITLE
Add is_blind function and replace checks

### PR DESCRIPTION
**Pull request by Night-Pryanik**

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -1957,3 +1957,11 @@ bool Character::made_of( const material_id &m ) const {
     static const std::vector<material_id> fleshy = { material_id( "flesh" ), material_id( "hflesh" ) };
     return std::find( fleshy.begin(), fleshy.end(), m ) != fleshy.end();
 }
+
+bool Character::is_blind() const
+{
+    if( worn_with_flag( "BLIND" ) || has_effect( effect_blind ) ) {
```
Comment (illi-kun) [inline src/character.cpp]: You can simplify it to
```return ( worn_with_flag( "BLIND" ) || has_effect( effect_blind ) );```

Comment (codemime) [inline src/character.cpp]: Also, I wonder if `u.has_active_bionic("bio_blindfold")` can become a part of the condition. It slightly modifies the logic of the code below, but seems more relevant to me.

Comment (illi-kun) [inline src/character.cpp]: @codemime I see some conditions where ```has_active_bionic("bio_blindfold")``` isn't checked or checked for the inverted state, so I think it's good as it is.

**Comment by Night-Pryanik** [comment]
@codemime I think you're right, `bio_blindfold` should be in is_blind.

**Comment by illi-kun** [comment]
Yeah, sorry all, it seems I've misread some conditions.

@Night-Pryanik feel free to PR the required changes.

**Pull request merged**